pkg/clusteragent/autoscaling/workload/external/recommender.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2025-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package external
+
+import (
+	"context"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling/workload"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling/workload/model"
+	le "github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/leaderelection/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	pollingInterval              = 30 * time.Second
+	externalRecommenderID string = "extr"
+)
+
+// Recommender is the interface used to fetch external recommendations
+type Recommender struct {
+	recommenderClient *recommenderClient
+	store             *autoscaling.Store[model.PodAutoscalerInternal]
+}
+
+// NewRecommender creates a new Recommender to start fetching external recommendations
+func NewRecommender(podWatcher workload.PodWatcher, store *autoscaling.Store[model.PodAutoscalerInternal]) *Recommender {
+	recommenderClient := newRecommenderClient(podWatcher)
+
+	return &Recommender{
+		recommenderClient: recommenderClient,
+		store:             store,
+	}
+}
+
+// Run starts the Recommender interface to generate external recommendations
+func (r *Recommender) Run(ctx context.Context) {
+	log.Debugf("Starting external autoscaling recommender")
+	ticker := time.NewTicker(pollingInterval)
+
+	go func() {
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				r.process(ctx)
+			case <-ctx.Done():
+				log.Debugf("Stopping external autoscaling recommender")
+				return
+			}
+		}
+	}()
+}
+
+func (r *Recommender) process(ctx context.Context) {
+	// Filter pod autoscalers to only retrieve autoscalers where external recommender is enabled
+	externalRecommendationFilter := func(dpa model.PodAutoscalerInternal) bool {
+		return dpa.CustomRecommenderConfiguration() != nil
+	}
+	podAutoscalers := r.store.GetFiltered(externalRecommendationFilter)
+
+	log.Debugf("Found %d pod autoscalers with external recommender enabled", len(podAutoscalers))
+
+	for _, podAutoscaler := range podAutoscalers {
+		// Fetch external recommendations
+		recommendation, err := r.recommenderClient.GetReplicaRecommendation(ctx, podAutoscaler)
+		telemetryHorizontalExternalRecommendations.Set(
+			float64(recommendation.Replicas),
+			podAutoscaler.Namespace(),
+			podAutoscaler.Spec().TargetRef.Name,
+			podAutoscaler.Name(),
+			string(recommendation.Source),
+			le.JoinLeaderValue,
+		)
+		r.updateAutoscalerAndUnlock(podAutoscaler.ID(), recommendation, err)
+		if err != nil {
+			log.Debugf("Got error fetching external recommendation for pod autoscaler %s: %v", podAutoscaler.ID(), err)
+		} else {
+			log.Debugf("Updated external recommendation for pod autoscaler %s", podAutoscaler.ID())
+		}
+	}
+}
+
+func (r *Recommender) updateAutoscalerAndUnlock(key string, horizontalRecommendation *model.HorizontalScalingValues, err error) {
```

Comment:
The lock and unlock being done in the same function, it's not necessary to advertise the `andUnlock` in the name.